include/zephyr/drivers/ipm.h

```diff
@@ -126,9 +126,10 @@ __subsystem struct ipm_driver_api {
  * data read back.
  *
  * @param ipmdev Driver instance
- * @param wait If nonzero, busy-wait for remote to consume the message. The
- *	       message is considered consumed once the remote interrupt handler
- *	       finishes. If there is deferred processing on the remote side,
+ * @param wait If true (!= 0), busy-wait indefinitely for the remote to consume
```

Comment:
`If non-zero, ...` would probably be better. Although it acts like a boolean, it is not actually a boolean variable. Saying true/false might confuse someone into passing `true`/`false` instead of integers.